### PR TITLE
Major test_va_api refactoring (Part 2)

### PR DESCRIPTION
**Pull request by uartie**
Refactored profile/entrypoint VAAPIFixture methods and updated tests accordingly.  Also moved common test case param setup to each test fixture class and implemented SetUp/TearDown methods.  This simplifies test logic a great deal.

### CODE REVIEW

File: test/test_va_api_createsurfaces.cpp [hunk 1]
```diff
@@ -157,45 +154,29 @@ class VAAPIQuerySurfaces
...
-    } else {
-        // profile not supported
+    if (not isSupported(profile, entrypoint)) {
         doLogSkipTest(profile, entrypoint);
+        return;
     }
+
+    createConfig(profile, entrypoint);
+    SurfaceAttributes attribs;
+    querySurfaceAttributes(attribs);
```
Comment (jayesht) [inline test/test_va_api_createsurfaces.cpp]: destroyConfig() is missing.

**Comment by jayesht** [comment]
LGTM.

File: test/test_va_api_createsurfaces.cpp (hunk 1 above)
Comment (uartie) [reply test/test_va_api_createsurfaces.cpp]: Good catch.  I will fix.

**Comment by uartie** [comment]
@jayesht updated. thanks.

**Comment by uartie** [comment]
@xhaihao any comments.  Once this series is merged I have another small set of patches ready to submit that should address most of the remaining issues reported by @XinfengZhang

**Pull request merged**